pykg2tbl/service.py

```diff
@@ -1,33 +1,41 @@
-# Use this file to describe the datamodel handled by this module
-# we recommend using abstract classes to achieve proper service and interface
-# insulation
 import csv
 import logging
 from abc import ABC, abstractmethod
+from typing import List, Tuple, Union
 
+import pandas as pd
 from rdflib import Graph
 from SPARQLWrapper import SPARQLWrapper
 
 log = logging.getLogger(__name__)
 
 
-class QueryResult:
+class QueryResult(ABC):
     """
     Class that incompases the result from a performed query
 
-    :param data: query result data in the form of a list of dictionaries
+    :param list data: query result data in the form of a list of dictionaries
+    :param str query: query
+
     """
 
-    def __init__(self, data: dict):
-        log.info(data)
-        self._data = data  # TODO pandas dataframe
+    def __init__(self, data: list, query: str = ""):
+        # log.info(data)
+        self._data = data
+        self.query = query
 
```

Comment:
Oh got it! 
I was totaly not thinking about subclassing QueryResult,  I think it could be just its own class, and the diferent methods to get what you need.

The input into this class is a standardized object, so It can use that default to be able to pursue multiple options